imap_processing/ena_maps/ena_maps.py

```diff
@@ -0,0 +1,536 @@
+        The Ultra instrument team has determined that all pixels in the map must
+        receive data, at least for calculating flux. So we will need to use the "pull"
+        method of index matching, where the map grid is projected to the pointing set
+        frame and binned to the edges of the pointing set grid.
+        # TODO: Implement the "pull" method of index matching.
+        """
+        if method == IndexMatchMethod.PUSH:
+            if pointing_set.reference_frame != self.reference_frame:
+                pointing_set.project_to_frame(self.reference_frame)
+
+            az_indices = (
+                np.digitize(pointing_set.az_el_points[:, 0], self.sky_grid.az_bin_edges)
+                - 1
+            )
+            el_indices = (
+                np.digitize(pointing_set.az_el_points[:, 1], self.sky_grid.el_bin_edges)
+                - 1
+            )
+            flat_indices = np.ravel_multi_index(
+                multi_index=(az_indices, el_indices),
+                dims=(
+                    len(self.sky_grid.az_bin_midpoints),
+                    len(self.sky_grid.el_bin_midpoints),
+                ),
+                order=self.order,
+            )
+
+        else:
+            raise NotImplementedError(
+                "The 'pull' method of index matching is not yet implemented."
+            )
+        return flat_indices
+
+    def project_pset_values_to_map(
+        self,
+        pointing_set: PointingSet,
+        value_keys: list[tuple[str, IndexMatchMethod]] | None = None,
+    ) -> None:
+        """
+        Project a pointing set's values to the map grid.
+
+        Parameters
+        ----------
+        pointing_set : PointingSet
+            The pointing set containing the values to project to the map.
+        value_keys : list[tuple[str, IndexMatchMethod]] | None
+            The keys of the values to project to the map, and method of index matching.
+            Ex.: [("counts", IndexMatchMethod.PUSH), ("flux", IndexMatchMethod.PULL)]
+            data_vars named each key must be present, and of the same dimensionality in
+            each pointing set which is to be projected to the map.
+            Default is None, in which case all data_vars in the pointing set are used,
+            with the "push" method of index matching.
+
+        Raises
+        ------
+        ValueError
+            If a value key is not found in the pointing set.
+        """
+        if value_keys is None:
+            value_keys = [
+                (key, IndexMatchMethod.PUSH)
+                for key in pointing_set.data.data_vars.keys()
+            ]
+
+        for value_key, _ in value_keys:
+            if value_key not in pointing_set.data.data_vars:
+                raise ValueError(f"Value key {value_key} not found in pointing set.")
+
+        # Determine the indices of the map grid that correspond to the pointing set
+        matched_indices = self.match_pset_coords_to_indices(pointing_set)
```

Comment:
I think this could be cached for each "project_values" we do. The coordinate mapping is always going to be the same and we are going to want to transform many different quantities.

Reply:
I agree we don't want to recalculate the indices for each quantity! Can you elaborate on what you see as a good way to cache this? It belongs to the combination of a PSET with it's frame and time, and a Map, and so I'm not sure the best place for the index mapping to "live", aside from within this function. 

The way I'd been intending to handle projecting multiple quantities (e.g. exposure time, counts) is to pass them into this function as 'value_keys'. It then finds the matching indices once, and bins each value_key with those indices.